### ISSUE TITLE
Out-of-date Docs? signup and magiclink types are deprecated?

### ISSUE THREAD

**Issue by nate-oo** [supabase/supabase#27883]
# Out-of-date documentation?

Under **Verify and log in through OTP** it states that `signup` and `magiclink` types are deprecated.

Under **Change the Auth confirmation path** it says to use `type=signup`. (see step 6)

Under **Signing in with Magic Link** it says to use `type=magiclink`.

Just bringing to your attention in case they need to be updated. If it does need to be updated, I'm unsure if there are other places in the docs as well. These are just the spots I noticed after a quick look.  I only looked at Nextjs related docs.

Thanks!
Nate

## Links

These links pertain to the statements above.
[Verify and log in through OTP](https://supabase.com/docs/reference/javascript/auth-verifyotp)
[Change the Auth confirmation path](https://supabase.com/docs/guides/auth/server-side/nextjs?queryGroups=router&router=app)
[Signing in with Magic Link](https://supabase.com/docs/guides/auth/auth-email-passwordless?queryGroups=language&language=js#signing-in-with-magic-link)

## Describe the problem

Possible out-of-date docs

## Additional context

Screen shots below are in the same order as the links I mentioned above.
<img width="1263" alt="Screenshot 2024-07-09 at 11 13 00 AM" src="https://github.com/supabase/supabase/assets/123698607/68dea01a-751a-4aa6-a4a6-a53da9d116bd">
<img width="1210" alt="Screenshot 2024-07-09 at 11 13 33 AM" src="https://github.com/supabase/supabase/assets/123698607/37178036-a23d-4407-afc5-c132f9e86ad5">
<img width="1600" alt="Screenshot 2024-07-09 at 11 18 52 AM" src="https://github.com/supabase/supabase/assets/123698607/bdf2879d-3a52-45ce-93bf-efcf82775879">

**Comment by thenbe** [comment]
Is (or will) `magiclink` actually deprecated though? If it is, what is it deprecated in favor of?

As far as I can tell, the first (and only?) mention of `magiclink` being deprecated is in this [PR](https://github.com/supabase/supabase/pull/27639) ([commit](https://github.com/supabase/supabase/commit/466458762479b2b5c2495d7cb0ad91f44c805efd#diff-4f871b20303236df420c42a8e6cfbb75499aac1a905779eb4757d13610becbefR957)). The diff does not state explicitly state what what `magiclink` is deprecated in favor of. But given that `email` is the only new added type within that line [diff](https://github.com/supabase/supabase/commit/466458762479b2b5c2495d7cb0ad91f44c805efd#diff-4f871b20303236df420c42a8e6cfbb75499aac1a905779eb4757d13610becbefR957), perhaps the author (hi @silentworks) intended to mean that `magiclink` was deprecated in favor of `email`?

**Comment by silentworks** [comment]
You are correct @nate-oo, I think these docs were written before `magiclink` and `signup` was deprecated. The correct type is `email` as you've stated. I will open a PR shortly with a fix for this. 

@thenbe yes that's correct, `magiclink` and `signup` are deprecated in favor of `email`. However for backward compatibility reasons `magiclink` and `signup` still works.

**Comment by silentworks** [comment]
This has been fixed now, the PR got merged in yesterday.

**Comment by Hallidayo** [comment]
Thank you @silentworks!
